src/arrayMethodJoin.js

```diff
@@ -5,7 +5,24 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    if (this.length === 0) {
+      return '';
+    }
+
+    const sep = separator === undefined ? ',' : String(separator);
+
+    let result = this[0] === null ? '' : String(this[0]);
+
+    for (let i = 1; i < this.length; i++) {
+      if (this[i] === undefined
+        || this[i] === null) {
+        result += sep;
+      } else {
+        result += sep + String(this[i]);
+      }
+    }
```

Comment:
So steps are next:
check if `this[i]` not equal `undefined` or `null` - add to result
check if `i` is not last - add separator
```suggestion
    for (let i = 0; i < this.length; i++) {
      const value = this[i];
      
      if (value !== undefined   && value !== null) {
        result += value;
      } 
      
      if (i !== this.length - 1) {
        result  += separator;
      }
     
    }
```